### ISSUE TITLE
[RoC] [GT5U] GregTech Aluminium Ingots do not work in Blast Furnace for producing Aluminum Alloy

### ISSUE THREAD

**Issue by Barhandar** [ReikaKalseki/Reika_Mods_Issues#1621]
The NEI recipe cycles through RoC aluminum, TiC aluminum, LibVulpes aluminum, and Immersive Engineering aluminium. The OD entries on all of them, including GT aluminium, are identical: `ingotAluminium`, `ingotAluminum`, `ingotNaturalAluminum`.

The blast furnace was friction heated, to 1002C (recipe requirement is 900C), didn't start processing (and WAILA indicated invalid recipe) until GT ingot was replaced with RoC ingot.

**Comment by ReikaKalseki** [comment]
Does that recipe even support the OreDict?

**Comment by Barhandar** [comment]
I don't know, it cycles through ingots suggesting it does, but code might be using a static list. I've just tested in singleplayer: TiC, IE and LibVulpes ingots do work, GT's one does not. Funnily, _ElectriCraft_'s aluminum ingot does not work either.

Additionally GregTech doesn't even recognize the existence of aluminum in RoC and ElC (or indeed in any other mod - only "aluminIum" entries are present in the relevant file), that is, there isn't an unification target option for it, which might be missing-material-related. However, it does somehow find a single ingotAluminium in DragonAPI.

**Comment by ReikaKalseki** [comment]
> it cycles through ingots suggesting it does

Then it does. It is possible it is being added to the OreDict after the recipe is created.

**Comment by OvermindDL1** [comment]
GT oredict entries are added to the oredict list in the `init` stage, which should work fine as recipes should only ever be added in the `postinit` stage (just like items/blocks and so forth should only ever be instanced/registered in the `preinit` stage).

GT also registers both aluminium (and all variations like ingotAluminium and such) as well as the wrong spelling of `aluminum` (like wtf on that...) and works with both spellings as well.

**Comment by Barhandar** [comment]
>as well as the wrong spelling of

Welcome to the world of platinium, stannium, cuprium, uranum, titanum, stibum, indum, cadmum, et cetera.   
That y'all scientists can't bloody agree which one of two equally valid spellings to use doesn't make one of them wrong.

>and works with both spellings as well.

Also, that means GT5U'd allow unification into RoC's ingots, but it does not.

**Comment by OvermindDL1** [comment]
> That y'all scientists can't bloody agree which one of two equally valid spellings to use doesn't make one of them wrong.

Well for things that appear on the periodic table of the elements it makes sense to use the actual name that is on that table.  ;-)

**Comment by ReikaKalseki** [comment]
> as recipes should only ever be added in the postinit stage

Tell that to 95% of mods, including mine. Almost every mod in existence registers them during main init.

**Comment by ReikaKalseki** [comment]
The easiest fix for this is to make RC load after GT. What is its modID?

**Comment by OvermindDL1** [comment]
As I recall:

For GT5:  `gregtech`

EDIT:  Fixed post based on lower information.  ^.^

**Comment by GregoriusT** [comment]
Uhhm that won't work, I already load GT6 after RoC. And my OreDict Stuff is done in the @Init Phase BEFORE all other Mods load their @Init (that is hacked to be ALWAYS the case)
my Mod consists out of 2 Parts, one that loads after all stuff and one that loads before all stuff, and the one that does the before stuff also does the @Init OreDict registering,
so that wouldn't fix for GT6. I guess this is an All-GT5U-Bug and works fine in GT6 therefore. I would need someone to confirm that it works for GT6 just fine.

**Comment by OvermindDL1** [comment]
Good thing I asked to confirm!  ^.^;

So only GT5u needs to be hacked/fixed.  ^.^

**Comment by GregoriusT** [comment]
What I mean with this is, do not use "after:gregapi_post", because it will crash both Mods.

Also it should work just fine with GT6, only GT5U is broken. I would recommend @BloodAsp (I hope I pinged him right) to just unificate Aluminium to the RotaryCraft Ingot to fix this Issue.

**Comment by OvermindDL1** [comment]
He has a hyphen in his name, thus:  @Blood-Asp  :-)

**Comment by ReikaKalseki** [comment]
> my Mod consists out of 2 Parts, one that loads after all stuff and one that loads before all stuff, and the one that does the before stuff also does the @init OreDict registering,

Then why is my recipe not accepting GT aluminum? It should be in the lists by the time the blast furnace is reading them (it is internally just calling `OreDictionary.getOres`).

**Comment by OvermindDL1** [comment]
> Then why is my recipe not accepting GT aluminum?

It's probably a GT5u bug, not a GT6 bug.  Have you tested if it works in GT6 or just GT5u?  GT5u is not managed by Gregorious but rather by Blood-ASP.

**Comment by ReikaKalseki** [comment]
I do not have GT - of any version - in my pack.

**Comment by Blood-Asp** [comment]
Will look at this from GT side. But certainly strange.

Another strange thing i heard about the last few weeks is that Lubricant (From GT or other mods using ID "lubricant") turns into RoC lubricant (ID "rc_lubricant") on server restart. Is there anything you can think off in RoC @ReikaKalseki or should i test all ideas i have on my side?

**Comment by GregoriusT** [comment]
Yeah I wondered where that came from or how that even works. Happens with GT6 too, but my solution was "just make both options work" so I don't have any Issues in GT6 with it, since my Stuff can always use both options and convert between them when needed.

**Comment by Blood-Asp** [comment]
make both work as in "add recipes for both" or as in "see them as the same fluid internally"?

**Comment by GregoriusT** [comment]
Added Recipes for both and added a "Generifier" Recipe for turning regular Lube into RoC Lube and vice versa.

The Generifier is just a Machine that turns things into identical or lower quality things for free and instantly (within 1 tick). I use it for example to turn all the different kinds of Juice into "Forestry Juice" for example, or to turn any sort of Wooden Plank into "The Generic Wooden Plank" so that it is stackable. It can also turn HSLA Steel into regular Steel (but not regular Steel into HSLA), to give an example on how it improves compatibility a lot.

**Comment by ReikaKalseki** [comment]
@Blood-Asp No, I have no system to unify fluids, nor would I want one.

**Comment by OvermindDL1** [comment]
> @Blood-Asp No, I have no system to unify fluids, nor would I want one.

If it happens only on server restart that does not sound like a unification but rather it sounds like the ID mappings in the fluid registry being borked some-how (different fluids sharing one ID some how?!).

**Comment by Barhandar** [comment]
>If it happens only on server restart that does not sound like a unification but rather it sounds like the ID mappings in the fluid registry being borked some-how (different fluids sharing one ID some how?!).

Or something stupid like partial-matching the ID-string, with RoC fluids being lower ID and thus getting picked first. Though if they're strictly partial-matching, there would also be issues with fuel-rc jet fuel and rc ethanol-bioethanol pairs.

**Comment by Blood-Asp** [comment]
It does not only happen on restart but also when filling into Extra Cells Certus tanks. But on restart it happens with every forge fluid storage. both tanks in world or universal fluid storage in inventorys.

**Comment by ReikaKalseki** [comment]
> Or something stupid like partial-matching the ID-string

> But on restart it happens with every forge fluid storage

This reminded me of something I implemented a long time ago, but had completely forgotten about until now:

Back in v7 I changed the fluid registry names of the RC/ReC fluids, prepending "rc " to their names to avoid conflicts. To avoid voiding massive quantities of fluid, on my dev server (the same one as in the most recent world tour) and in others' saves, I implemented a system where the old fluids would be converted to my fluids on NBT load.

**However**, this was entirely within my own tanks' handling system, making it not a suspect for an issue involving all tanks. And then I found [this](https://github.com/ReikaKalseki/DragonAPI/blob/master/ASM/FluidNamePatch.java).

As it turns out, there is a bug in the converter; it is only supposed to convert the fluid if there is no mapping for the old name.

So yes, this *is* my fault, and will be fixed for v19, which is not far away.
